### DISCUSSION TITLE
Units strings for time units are tricky

### DISCUSSION THREAD

**Discussion by pp-mo** [SciTools/cf-units#369] [General]
The problem is calendars, really.

For most units, `Unit(str(my_unit)) == my_unit`
But with (nonstandard) calendars ...
  1. the calendar does not appear in the printed form (i.e. the "unit string" or string form of the unit)
  2. there is no way of including it, since the syntax does not support it.
  
E.G. (1.)
```
>>> ut = Unit('days since 2001-01-01', calendar='360_day') 
>>> ut
Unit('days since 2001-01-01', calendar='360_day')
>>> print(ut)
days since 2001-01-01
```
I.E. the calendar is lost in producing a 'unit string' form.

E.G. (2.)
```
>>> Unit('days since 2001-01-01, calendar=360_day')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/tmp/persistent/newconda-envs/ncdata/lib/python3.11/site-packages/cf_units/__init__.py", line 830, in __init__
    raise value_error from None
ValueError: [UT_SYNTAX] Failed to parse unit "days since 2001-01-01, calendar=360_day"
```
Obviously, only one "possible form", but I think **nothing** like this is valid.

According to [UDUNITS docs](https://docs.unidata.ucar.edu/udunits/current/udunits2lib.html#Time) ...
> You should use a true calendar package rather than the UDUNITS-2 package to handle time. ... the hybrid Gregorian/Julian calendar used by the UDUNITS-2 package _cannot be changed_.

**Comment by pp-mo** [comment]
There is more general issue here, of how much the user can expect to rely on the equivalence of a unit and its "string form".

We naturally have lots of cases where, for two units ut1 and ut2 :  `ut1 == ut2` but also `str(ut1) != str(ut2)`
This however is perfectly natural because == uses an equivalence concept.
E.G. `Unit('1.0km') == Unit('1000.0m')`

For time units, the time/date representation is particularly fluid.
E.G. 
```
>>> ut1 = Unit('days since 1970')
>>> ut2 = Unit('24.0 hours since 1970-01-01T00:00:00')
>>> ut1
Unit('days since 1970', calendar='standard')
>>> ut2
Unit('24.0 hours since 1970-01-01T00:00:00', calendar='standard')
>>> str(ut1)
'days since 1970'
>>> str(ut2)
'24.0 hours since 1970-01-01T00:00:00'
>>> ut1 == ut2
True
```

**Comment by ESadek-MO** [comment]
@SciTools/peloton agrees that `str(ut2)` could produce the calendar, as in `>>> ut2`, we should make an issue for this (in cases where calendar is non-standard). This would cause a break in udunits unit creation, but we think that's desired behavior.

**Reply by ESadek-MO** [reply]
@pp-mo might be the person for this?
